File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.swift
import Foundation

var result = ""
var changeCount = 0
var removeCount = 0

func solution(_ s:String) -> [Int] {
    result = s
    
    while result != "1" {
        var str = removeZero(str: result)
        result = String(str.count, radix: 2)
        changeCount += 1
    }
    
    return [changeCount, removeCount]
}

func removeZero(str: String) -> String {
    return str.map { ch in
        if ch == "1" {
            return "1"
        } else {
            removeCount += 1
            return ""
        }
    }.joined()
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int changeCount = 0;
int removeCount = 0;

int getOneCount(string s) {
    int result = 0;

    for (char ch : s) {
        (ch == '1') ? result++ : removeCount++; 
    }

    return result;
}

string toBinary(int num) {
    string result = "";

    while(num != 1) {
        result += to_string(num % 2);
        num /= 2;
    }

    result += to_string(num);

    reverse(result.begin(), result.end());

    return result;
}

vector<int> solution(string s) {
    vector<int> answer;

    while (s != "1") {
        int size = getOneCount(s);
        s = toBinary(size);
        changeCount++;
    }

    answer.push_back(changeCount);
    answer.push_back(removeCount);

    return answer;
}